src/scraper/scrape.ts

```diff
@@ -6,9 +6,56 @@ import {
 import { AccountConfig } from "../types.js";
 import { ScraperErrorTypes } from "israeli-bank-scrapers/lib/scrapers/errors.js";
 import { createLogger } from "../utils/logger.js";
+import { config } from "../config.js";
+import { requestOtpCode } from "../bot/notifier.js";
 
 const logger = createLogger("scrape");
 
+/**
+ * Creates an OTP code retriever function for OneZero accounts
+ */
+function createOtpCodeRetriever(
+  companyId: string,
+  phoneNumber: string,
+): () => Promise<string> {
+  return async () => {
+    if (!config.options.notifications.telegram?.enableOtp) {
+      throw new Error("OTP is not enabled in configuration");
+    }
+
+    logger(
+      `Requesting OTP code for ${companyId} account (phone: ${phoneNumber.substring(0, 4)}...)`,
+    );
+    return await requestOtpCode(companyId, phoneNumber);
+  };
+}
+
+/**
+ * Prepares the account credentials with OTP support if needed
+ */
+function prepareAccountCredentials(account: AccountConfig): AccountConfig {
+  // Check if this is a OneZero account that needs OTP
+  if (
+    account.companyId === "oneZero" &&
+    config.options.notifications.telegram?.enableOtp &&
+    "phoneNumber" in account &&
+    account.phoneNumber &&
+    !("otpLongTermToken" in account)
+  ) {
```

Comment:
This logic can be moved to a `shouldCreateOtpRetriver` and be placed in a new otp file